scripts/qa_crowbarsetup.sh

```diff
@@ -3421,10 +3441,97 @@ function onadmin_run_cct()
     return 0
 }
 
+# Set the aliases for nodes.
+# This is usually needed before batch step, so batch can refer
+# to node aliases in the scenario file.
+function onadmin_setup_aliases()
+{
+    local nodesavailable=`get_all_discovered_nodes`
+    local i
+
+    if [ -n "$want_node_aliases" ] ; then
+        # aliases provided explicitely, assign them successively to the nodes
+        # example: want_node_aliases=controller=1,swift=2,kvm=2
+
+        for aliases in ${want_node_aliases//,/ } ; do
+
+            # split off the number => group
+            node_alias=${aliases%=*}
+            # split off the group => number
+            number=${aliases#*=}
+
+            i=1
+            for node in `printf  "%s\n" $nodesavailable | head -n$number`; do
+                if [[ $i -gt 1 ]]; then
+                    node_alias="$node_alias$i"
+                fi
+                set_node_alias $node $node_alias
+                nodesavailable=`printf "%s\n" $nodesavailable | grep -iv $node`
+                i=$((i+1))
+            done
+        done
+    else
+        # try to setup aliases automatically
+        if [[ $hacloud = 1 ]] ; then
+            # 1. HA
+            # use the logic from cluster_node_assignment and assign
+            #      dataN, serviceN, networkN aliases
+            # for nodes in clusternodesdata etc.
+
+            cluster_node_assignment
+
+            local i=1
+            for node in $clusternodesdata ; do
+                set_node_alias $node "data$i"
+                i=$((i+1))
+            done
+            i=1
+            for node in $clusternodesnetwork ; do
+                set_node_alias $node "network$i"
+                i=$((i+1))
+            done
+            i=1
+            for node in $clusternodesservices ; do
+                set_node_alias $node "services$i"
+                i=$((i+1))
+            done
```

Comment:
3x code duplication - not so nice... would probably need an eval to handle the different variable names
```bash
for x in data network services ; do
  eval "for node in \$custernodes$x ; do
    set_node_alias \$node \"$x\$i\"
    ...
  done" ; 
done
```